src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java

```diff
@@ -115,20 +115,29 @@ public void EntityJoinWorld(final EntityJoinWorldEvent event) {
         }
     }
     
+    @SubscribeEvent
+    public void Respawn(cpw.mods.fml.common.gameevent.PlayerEvent.PlayerRespawnEvent event) {
+        // Exit Portal fix
+        if (!event.player.worldObj.isRemote) {
```

Comment:
No. At least check if you are indeed returning from the end or just happened to have died.

Reply:
Is this a huge fix or just a bit code change ? @Glease

Reply:
The relevant info (return from end?) seems not be passed via this fml event. If we are able to get it from other means then all good, otherwise it will quickly become tedious, i.e. require injecting a new call hook into the relevant minecraft code via asm/mixin

Reply:
So better close this fix or just pulled the first one? @Glease

Reply:
Invalid concern. This is fine to merge.